packages/beacon-node/src/execution/engine/interface.ts

```diff
@@ -147,5 +174,13 @@ export interface IExecutionEngine {
 
   getPayloadBodiesByRange(start: number, count: number): Promise<(ExecutionPayloadBody | null)[]>;
 
+  getClientVersion(clientVersion?: ClientVersion): Promise<ClientVersion[]>;
+
   getState(): ExecutionEngineState;
+  /**
+   * Not to be confused with `getClientVersion`. This method returns cached client version
+   * from `getClientVersion` which is a rpc call to EL client
+   */
+  getExecutionClientVersion(): ClientVersion[];
```

Comment:
instead of using a function for this, could have a public property and use a getter in the implementation

Reply:
> instead of using a function for this, could have a public property and use a getter in the implementation

It was to conform with the `getState()` function. But I agree using a getter. We don't need property to be public though.